### PR TITLE
Fix setting enqueuedUploadJobsWh fix enqueuing jobs

### PR DESCRIPTION
**Pull request by psrikanth88**
**Fixes** # (*issue*)

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
## Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by SumanthPuram — APPROVED** [review]

File: warehouse/slave.go [hunk 1]
```diff
@@ -135,51 +136,69 @@ func (job *PayloadT) getColumnName(columnName string) string {
...
-		tableName := tableName
-		outputFile := outputFile
-		maxParallelLoadsChan <- struct{}{}
-		rruntime.Go(func() {
-			uploadOutput, err := jobRun.uploadLoadFileToObjectStorage(uploader, &outputFile, tableName)
-			if err != nil {
-				loadFileErrChan <- err
-				return
+	loadFileIDChan := make(chan int64, len(jobRun.outputFileWritersMap))
+	uploadJobChan := make(chan *loadFileUploadJob, 100)
```
Comment (snarkychef) [inline warehouse/slave.go:152]: Why 100? it should be len(jobRun.outputFileWritersMap) ??

File: warehouse/slave.go [hunk 2]
```diff
@@ -135,51 +136,69 @@ func (job *PayloadT) getColumnName(columnName string) string {
...
+	uploadErrorChan := make(chan error)
+	ctx, cancel := context.WithCancel(context.Background())
+	for i := 0; i < numLoadFileUploadWorkers; i++ {
+		go func(ctx context.Context) {
+			for uploadJob := range uploadJobChan {
+				select {
+				case <-ctx.Done():
+					pkgLogger.Debugf("context is cancelled, stopped processing load file for staging file id %s ", job.StagingFileID)
```
Comment (snarkychef) [inline warehouse/slave.go:160]: We should use Errorf here?

File: warehouse/slave.go [hunk 3]
```diff
@@ -135,51 +136,69 @@ func (job *PayloadT) getColumnName(columnName string) string {
...
-		outputFile := outputFile
-		maxParallelLoadsChan <- struct{}{}
-		rruntime.Go(func() {
-			uploadOutput, err := jobRun.uploadLoadFileToObjectStorage(uploader, &outputFile, tableName)
-			if err != nil {
-				loadFileErrChan <- err
-				return
+	loadFileIDChan := make(chan int64, len(jobRun.outputFileWritersMap))
+	uploadJobChan := make(chan *loadFileUploadJob, 100)
+	uploadErrorChan := make(chan error)
```
Comment (snarkychef) [inline warehouse/slave.go:153]: We are using unbuffered channel here. We need to understand the behavior when multiple upload workers throw errors. We only consume the first error from this channel and main go routine exits from the function. And other uploader go routines will be blocked in sending error to this channel. 
Maybe we can just use a buffered channel of length upload worker size here?

**Review by snarkychef — CHANGES_REQUESTED** [review]

File: warehouse/slave.go (hunk 1 above)
Comment (gane5hvarma) [reply warehouse/slave.go:152]: forgot to revert this

File: warehouse/slave.go (hunk 2 above)
Comment (gane5hvarma) [reply warehouse/slave.go:160]: we are logging error when we are receiving error from the err channel

File: warehouse/slave.go (hunk 3 above)
Comment (gane5hvarma) [reply warehouse/slave.go:153]: updated to buffered channel

**Review by snarkychef — APPROVED** [review]

**Pull request merged**